modules/programs/bash.nix

```
@@ -190,7 +190,10 @@ in {
         HISTCONTROL = concatStringsSep ":" cfg.historyControl;
       } // optionalAttrs (cfg.historyIgnore != [ ]) {
         HISTIGNORE = escapeShellArg (concatStringsSep ":" cfg.historyIgnore);
-      }));
+      }))) + (optionalString (cfg.historyFile != null) ''
+
+        mkdir -p "$(dirname "$HISTFILE")"
+      '');
```

Maybe sufficient to do

```suggestion
      }) ++ optional (cfg.historyFile != null)
        ''mkdir -p "$(dirname "$HISTFILE")"'');
```

---

much simpler indeed. Thank you.